lib/Dialect/AIEVec/Transforms/VectorToVectorConversions.cpp

```diff
@@ -788,6 +857,8 @@ void xilinx::aievec::buildCanonicalizeVectorForAIEVec(
   // Add `Vector` code canonicalization passes
   // TODO: Add passes to unroll vector with unsupported types
   // TODO: Add passes to split vectors that won't fit in registers
+  if (decodeTargetBackend(options.targetBackend) == TargetBackend::LLVMIR)
```

Comment:
What will break when the backend is C++?

Reply:
It will likely be immediately undone by the other pattern we have, the one that hoists `ext` ops up to the load/block argument. It's disabled for the LLVMIR backend because it was breaking pattern matching in certain circumstances.

Reply:
I see. Thanks for the explanation.